ROX-27350: Setup ACS trusted tasks builds and publishing

msugakov opened the pull request:
## Notes

* This change can be reviewed per commits as I broke it down in chunks and put descriptions that seemed meaningful. However, due the large number of commits, it can be quicker to just review the final version.
* I know that duplication in `determine-image-tag` isn't great but it will not be the right time to address that now. Let's leave that for ROX-27384 or, most likely, ROX-26026.
* Some tasks in the build pipeline don't feel quite fine, e.g. `clair-scan` and `clamav-scan` but they aren't failing the pipeline either, and so I decided to keep them in the hopes that one day they may start working. By keeping the tasks it will hopefully be easier to conform EC when that will be required.

## Links

* Goes on top of the initial pipelines setup https://github.com/stackrox/konflux-tasks/pull/2 (already merged).
* Used in
	* https://github.com/stackrox/stackrox/pull/13599
	* https://github.com/stackrox/scanner/pull/1742
	* https://github.com/stackrox/collector/pull/1992
* Created these Quay repos:
    * [quay.io/rhacs-eng/konflux-tasks](https://quay.io/repository/rhacs-eng/konflux-tasks?tab=tags) for tasks themselves.
    * [quay.io/rhacs-eng/konflux-tasks-trust](https://quay.io/repository/rhacs-eng/konflux-tasks-trust?tab=tags) for EC.


## Testing

- Tested builds in the consuming repos are fine with the new tasks. See the PRs linked above.
- EC status is hard to present due to known issues of EC logs disappearing but I'm confident from earlier testing that all should be good once we update the policy.

msugakov commented:
/retest acs-konflux-tasks-on-push

msugakov commented:
/retest acs-konflux-tasks-on-push

msugakov commented:
/retest

tommartensen commented on .tekton/acs-konflux-tasks-push.yaml:
```diff
@@ -74,52 +82,87 @@ spec:
+
+    workspaces:
+    - name: git-auth
+    - name: netrc
```
Do we need this workspace? We don't use it in our other pipelines.

msugakov replied:
`netrc` came from auto-generated pipelines. Our older ones were generated without it.

As I understand, it is a mechanism to add custom authentication for `curl` in prefetch task. Since we don't use it and are unlikely to start using soon, I deleted it.

tommartensen reviewed (approved):
LGTM, see comments.
One more thing: How do we expect the images used in the tasks to be kept up to date?

tommartensen commented on .tekton/acs-konflux-tasks-push.yaml:
```diff
@@ -328,40 +294,18 @@ spec:
-        - "false"
+        values: [ "false" ]
+
     - name: sast-snyk-check
```
> Some tasks in the build pipeline don't feel quite fine, e.g. clair-scan and clamav-scan but they aren't failing the pipeline either, and so I decided to keep them in the hopes that one day they may start working. By keeping the tasks it will hopefully be easier to conform EC when that will be required.

answered in PR description!

msugakov commented:
>  One more thing: How do we expect the images used in the tasks to be kept up to date?

Since this is a Component repo, I expect Renovate should start working right away. There is an issue though that tasks are in a non-standard directory (`tasks/`) where the config [expects](https://github.com/konflux-ci/mintmaker/blob/main/config/renovate/renovate.json#L69) `.tekton/`.
I was going to put our custom renovate config to the repo as part of <https://issues.redhat.com/browse/ROX-27352>, and I'll make sure to add `tasks/` dir to the Tekton manager.

Another problem is updating usage of these tasks in StackRox, Collector, and Scanner repos. As I understand, the vanilla config wouldn't recognize our tasks because it has this filter
```json
        "matchPackageNames": [
          "/^quay.io/redhat-appstudio-tekton-catalog//",
          "/^quay.io/konflux-ci/tekton-catalog//"
        ],
```
What this means is that we'll need to customize renovate config in StackRox repos to also pick up repos of our packages. To me it's part of the same <https://issues.redhat.com/browse/ROX-27352>.

msugakov commented:
/retest acs-konflux-tasks-on-push

msugakov commented:
/retest acs-konflux-tasks-on-push

msugakov commented:
/retest acs-konflux-tasks-on-push

msugakov commented:
There was some failure during container build that seemed like a bug. It got resolved after I manually deleted a tag from Quay repo. Let's keep an eye on this in the future.

```
step-build
*Warning*: This is an experimental command, it's usage and behavior can change in the next release(s)
Creating Tekton Bundle:
	- Added Task: fetch-external-networks to image
	- Added Task: retag-image to image
	- Added Task: fetch-scanner-v4-vuln-mappings to image
	- Added Task: fetch-scanner-v2-data to image
	- Added Task: determine-image-tag-stackrox to image
	- Added Task: determine-image-tag to image
	- Added Task: wait-for-image to image
Error: could not push image to registry as "quay.io/rhacs-eng/konflux-tasks:rev-019ec288c660e477572c6f36c4bcf4c39855b856": GET https://quay.io:443/v2/rhacs-eng/konflux-tasks/manifests/rev-019ec288c660e477572c6f36c4bcf4c39855b856: MANIFEST_UNKNOWN: manifest unknown; map[]
```

Affected
1. https://console.redhat.com/application-pipeline/workspaces/rh-acs/applications/acs-konflux-tasks/pipelineruns/acs-konflux-tasks-on-push-jxc8v
2. https://console.redhat.com/application-pipeline/workspaces/rh-acs/applications/acs-konflux-tasks/pipelineruns/acs-konflux-tasks-on-push-vnl75
